src/app/pages/send/send.component.tsx

```diff
@@ -227,6 +233,32 @@ export class SendPage extends Component<IProps, IState> {
         );
     }
 
+    public getAddressFromName(name) {
+        let address: string = name;
+
+        if (name.charAt(name.length - 4) === '.') {
+            const nameService = name.substring(name.length - 3);
+            const namicorn = new Namicorn({ debug: true, disableMatcher: true });
+            if (nameService === 'eth') {
```

Comment:
Idea: Maybe we could just do:

```js
if (nameService in namicorn.middleware) {
  namicorn.use(namicorn.middleware[nameService]());
}
```

This way, we don't have to be _so_ specific.